internal/grpc/certificates_test.go

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) F5, Inc.
+//
+// This source code is licensed under the Apache License, Version 2.0 license found in the
+// LICENSE file in the root directory of this source tree.
+package grpc
+
+import (
+	"crypto/tls"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/nginx/agent/v3/test/helpers"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	keyFileName        = "key.pem"
+	certFileName       = "cert.pem"
+	caFileName         = "ca.pem"
+	nonPemCaFileName   = "ca.nonpem"
+	nonPemCertFileName = "cert.nonpem"
+	certificateType    = "CERTIFICATE"
+	privateKeyType     = "RSA PRIVATE KEY"
+)
+
+var pathSeparator = string(os.PathSeparator)
+
+func TestAppendCertKeyPair(t *testing.T) {
+	tmpDir := t.TempDir()
+	key, cert := helpers.GenerateSelfSignedCert(t)
+
+	keyContents := helpers.Cert{Name: keyFileName, Type: privateKeyType, Contents: key}
+	certContents := helpers.Cert{Name: certFileName, Type: certificateType, Contents: cert}
+	certNonPemContents := helpers.Cert{Name: nonPemCertFileName, Type: "", Contents: cert}
+
+	keyFile := helpers.WriteCertFiles(t, tmpDir, keyContents)
+	certFile := helpers.WriteCertFiles(t, tmpDir, certContents)
+	nonPEMFile := helpers.WriteCertFiles(t, tmpDir, certNonPemContents)
+
+	testCases := []struct {
+		testName  string
+		certFile  string
+		keyFile   string
+		isError   bool
+		certAdded bool
+	}{
+		{
+			testName:  "valid files",
+			certFile:  certFile,
+			keyFile:   keyFile,
+			isError:   false,
+			certAdded: true,
+		},
+		{
+			testName: "no files",
+			certFile: "",
+			keyFile:  "",
+			isError:  false,
+		},
+		{
+			testName: "only cert file",
+			certFile: certFile,
+			keyFile:  "",
+			isError:  true,
+		},
+		{
+			testName: "only key file",
+			certFile: certFile,
+			keyFile:  "",
+			isError:  true,
+		},
+		{
+			testName: "missing cert file",
+			certFile: fmt.Sprintf("%s%s%s%s", pathSeparator, "invalid", pathSeparator, certFileName),
+			keyFile:  keyFile,
+			isError:  true,
+		},
+		{
+			testName: "missing key file",
+			certFile: certFile,
+			keyFile:  fmt.Sprintf("%s%s%s%s", pathSeparator, "invalid", pathSeparator, keyFileName),
+			isError:  true,
+		},
+		{
+			testName: "non PEM format cert",
+			certFile: nonPEMFile,
+			keyFile:  keyFile,
+			isError:  true,
+		},
```

Comment:
Do we need to check expired certs/keys?  There's some scenarios where user can upload a cert/key in a single file via PKCS

Reply:
I don't know if we supported that case before?

Reply:
I don't think we have before but wanted to check.